template.tpl

```diff
@@ -365,17 +402,28 @@ if (!data.enableFirstPartyCookies) {
 }
 
 
-var eventMetadata = {
-  currency: data.currency,
-  value: data.transactionValue,
-};
+var eventMetadata = {};
+
+if (data.currency) {
+  eventMetadata.currency = data.currency;
+}
+
+if (data.transactionValue) {
+  eventMetadata.value = data.transactionValue;
+}
+
+// If there is product information, add it to the eventMetadata. The simple table is
+// of the correct format - an array of objects.
+if (data.products && data.products.length) {
+  eventMetadata.products = data.products;
+}
 
 // Certain events don't support certain params, so we conditionally set them
-if (data.eventType != "AddToCart" && data.eventType != "AddToWishlist") {
+if (data.eventType != "AddToCart" && data.eventType != "AddToWishlist" && data.transactionId) {
   eventMetadata.transactionId = data.transactionId;
 }
 
-if (data.eventType != "SignUp" && data.eventType != "Lead") {
+if (data.eventType != "SignUp" && data.eventType != "Lead" && data.itemCount) {
```

Comment:
This is a bug.  A zero value is falsy but is meaningful.  We don't want to exclude it.